setup_bitswap.go

```diff
@@ -83,7 +83,9 @@ func setupBitswapExchange(ctx context.Context, cfg Config, h host.Host, cr routi
 
 type noopPeerLedger struct{}
 
-func (*noopPeerLedger) Wants(p peer.ID, e wl.Entry) {}
+func (*noopPeerLedger) Wants(p peer.ID, e wl.Entry) bool {
+	return false
+}
```

Comment:
@gammazero rainbow has this `noopPeerLogger` implementation and i had to start returning a `bool` after updating boxo here.

Seems that hardcoding `false` here causes `panic` in 
https://github.com/ipfs/boxo/blob/v0.22.0/bitswap/server/internal/decision/engine.go#L876

[`go test` log here](https://github.com/ipfs/rainbow/actions/runs/10287647684/job/28471187453?pr=158#step:13:265)

Thoughts? I can fix the panic by always returning `true` here, but suspect people who created their own noop PeerLogger (to disable provides and disable peer ledger entierely) may hit the same error, so we may want to address this edge case upstream.

Reply:
Removed `noopPeerLedger`.

Rationale:
- After looking at things bit deeper I realized `WithPeerBlockRequestFilter` is enough and custom `PeerLedger` does not help us with anything, and after overflow fixes in boxo [0.22](https://github.com/ipfs/boxo/releases/tag/v0.22.0) only causes problems (because we do want wantlist for wants that are not filtered by `WithPeerBlockRequestFilter`.
- Low risk since `RAINBOW_SEED_PEERING` is `false` by default, and it is marked as experimental feature (that we decided to not use in our infra).